source/renderer/zettlr-editor.js

```diff
@@ -503,6 +521,38 @@ class ZettlrEditor {
   }
   // END constructor
 
+  /**
+  * Computes and returns the width of a character in the monospace font in pixels.
+  * @see https://stackoverflow.com/a/118251/873661
+  */
+  computeMonospaceWidth () {
+    if (this._monospaceWidth > 0) return this._monospaceWidth
+
+    // We use 50 characters in order to get an approximately correct width (clientWidth is an integer)
+    var container = document.createElement('div')
+    container.id = 'measureMonoWidth'
+    container.innerHTML = '<span>--------------------------------------------------</span>'
+    this._cm.getWrapperElement().appendChild(container)
+    this._monospaceWidth = (container.clientWidth + 1) / 50
+    return this._monospaceWidth
+  }
+
+  /**
+  * Computes and returns the width of a character in the monospace font in pixels.
+  * @see https://stackoverflow.com/a/118251/873661
+  */
+  computeSpaceWidth () {
```

Comment:
This and the above method look very similar. Would it be possible to extract common code to a function?

Reply:
They are indeed similar, but have enough difference that I couldn't extract a meaningful common ancestor. Do you have a concert idea of how to achieve this?

Reply:
Maybe by introducing a function `measureCharWidth(divId, symbol)` — although I'm not sure why you use `-` for measuring spaces for the monospace font, when spaces should have the same width. When using `&nbsp;` for all, you'd have only one argument to the function.

Reply:
Good idea. I've now implemented it that way. 👍